src/sentry/snuba/metrics_layer/query.py

```diff
@@ -73,21 +129,23 @@ def bulk_run_query(requests: list[Request]) -> list[Mapping[str, Any]]:
     if not requests:
         return []
 
-    queries = []
-    for request in requests:
-        request, start, end = _setup_metrics_query(request)
-        queries.append([request, start, end])
+    referrer = requests[0].tenant_ids.get("referrer") or "unknown"
+    logging_tags = {"referrer": referrer, "lang": "mql"}
 
-    logging_tags = {"referrer": request.tenant_ids["referrer"] or "unknown", "lang": "mql"}
+    queries: list[list[Any]] = []
+    for request in requests:
+        updated_request, reverse_mappings, mappings = _resolve_metrics_query(request, logging_tags)
```

Comment:
Query resolution was moved before query setup in order to have access to the use case id when computing the granularity.